chain/index/ddls.go

```diff
@@ -0,0 +1,82 @@
+package index
+
+import "database/sql"
+
+const DefaultDbFilename = "chainindex.db"
+
+var ddls = []string{
+	`CREATE TABLE IF NOT EXISTS tipset_message (
+		message_id INTEGER PRIMARY KEY,
+		tipset_key_cid BLOB NOT NULL,
+		height INTEGER NOT NULL,
+		reverted INTEGER NOT NULL,
+		message_cid BLOB,
+		message_index INTEGER,
+		UNIQUE (tipset_key_cid, message_cid)
+	)`,
+
+	`CREATE TABLE IF NOT EXISTS eth_tx_hash (
+		tx_hash TEXT PRIMARY KEY,
+		message_cid BLOB NOT NULL,
+		inserted_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+	)`,
+
+	`CREATE TABLE IF NOT EXISTS event (
+		event_id INTEGER PRIMARY KEY,
+		message_id INTEGER NOT NULL,
+		event_index INTEGER NOT NULL,
+		emitter_addr BLOB NOT NULL,
```

Comment:
Flagging a discussion we're currently having sync regarding https://github.com/filecoin-project/lotus/issues/11594 suggesting that maybe we should store both the id and the delegated address - current code that figures out what to store here stores an ID address unless it's delegated in which case it stores the delegated address. So it may be better to just store the ID if it's not delegated and leave address null and then if it is delegated then store both.

Reply:
@rvagg Is there a valid use case to do it/what does it get us  ? We can always go from an id addr -> delegated addr.

Reply:
This is done in commit https://github.com/filecoin-project/lotus/pull/12421/commits/7b304a4b4e3b3924d70fecd01555b4157a4196bd.